src/components/checkbox/checkbox.js

```diff
@@ -171,12 +183,20 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
       }
 
       function render() {
-        if(ngModelCtrl.$viewValue) {
+        if(ngModelCtrl.$viewValue && !isIndeterminate) {
           element.addClass(CHECKED_CSS);
         } else {
           element.removeClass(CHECKED_CSS);
         }
       }
+
+      function setIndeterminateState(newValue) {
+        isIndeterminate = newValue !== false;
+        if (isIndeterminate) {
+          element.attr('aria-checked', 'mixed');
```

Comment:
It actually is set, and accounted for in the [tests](https://github.com/DerekLouie/material/blob/inteterminateCheckbox/src/components/checkbox/checkbox.spec.js#L43). It was a little curious though that the code isn't found in this file..